src/contrib/dependencyinjection/Akka.DI.Core/Extensions.cs

```diff
@@ -61,12 +66,40 @@ public static Type GetTypeValue(this string typeName)
         {
             var firstTry = Type.GetType(typeName);
             Func<Type> searchForType = () =>
-                AppDomain.CurrentDomain
-                    .GetAssemblies()
+                GetLoadedAssemblies()
                     .SelectMany(x => x.GetTypes())
                     .FirstOrDefault(t => t.Name.Equals(typeName));
-            
             return firstTry ?? searchForType();
         }
+
+        /// <summary>
+        /// Gets the list of loaded assemblies
+        /// </summary>
+        /// <returns>The list of loaded assemblies</returns>
+        private static IEnumerable<Assembly> GetLoadedAssemblies()
+        {
+#if APPDOMAIN
+            return AppDomain.CurrentDomain.GetAssemblies();
+#elif CORECLR 
+            var assemblies = new List<Assembly>();
+            var dependencies = DependencyContext.Default.RuntimeLibraries;
+            foreach (var library in dependencies)
+            {
+                try
+                {
+                    var assembly = Assembly.Load(new AssemblyName(library.Name));
+                    assemblies.Add(assembly);
+                }
+                catch
+                {
+                    //do nothing can't if can't load assembly
+                }
+            }
+            return assemblies;
+#else
```

Comment:
I'm thinking we don't need this `#else` directive since there will only either be `net45` or `netstandard1.6` and if anyone were to add another target they would need to make sure the .csproj is updated correctly to cover either `APPDOMAIN` or `CORECLR` conditionals.